scripts/prepare_data.py

```diff
@@ -62,67 +164,120 @@ def add_tile_id(row):
+            EPT_aoi_boundaries_df = EPT_aoi_4326_gdf.bounds
+
+            # Get tile coordinates and shapes
+            logger.info("- Tiling of the empty tiles AoI")  
+            empty_tiles_4326_all_gdf = aoi_tiling(EPT_aoi_boundaries_df)
+            # Delete tiles outside of the AoI limits 
+            empty_tiles_4326_aoi_gdf = gpd.sjoin(empty_tiles_4326_all_gdf, EPT_aoi_4326_gdf, how='inner', lsuffix='ept_tiles', rsuffix='ept_aoi')
+            # Attribute a year to empty tiles if necessary
+            if 'year' in labels_4326_gdf.keys():
+                if isinstance(EPT_YEAR, int):
+                    empty_tiles_4326_aoi_gdf['year'] = int(EPT_YEAR)
+                else:
+                    empty_tiles_4326_aoi_gdf['year'] = np.random.randint(low=EPT_YEAR[0], high=EPT_YEAR[1], size=(len(empty_tiles_4326_aoi_gdf)))
+            elif EPT_SHPFILE and EPT_YEAR: 
+                logger.warning("No year column in the label shapefile. The provided empty tile year will be ignored.")
+        elif EPT == 'shp':
+            if EPT_YEAR:
+                logger.warning("A shapefile of selected empty tiles are provided. The year set for the empty tiles in the configuration file will be ignored")
+                EPT_YEAR = None
+            empty_tiles_4326_aoi_gdf = EPT_aoi_4326_gdf.copy()
+
+    # Get all the tiles in one gdf 
+    if EPT_SHPFILE:
+        logger.info("- Concatenate label tiles and empty AoI tiles") 
+        tiles_4326_all_gdf = pd.concat([tiles_4326_labels_gdf, empty_tiles_4326_aoi_gdf])
+    else: 
+        tiles_4326_all_gdf = tiles_4326_labels_gdf.copy()
 
-    for label_boundary in label_boundaries_df.itertuples():
-        coords = (label_boundary.minx, label_boundary.miny, label_boundary.maxx, label_boundary.maxy)   
-        tiles_4326 = gpd.GeoDataFrame.from_features([tms.feature(x, projected=False) for x in tms.tiles(*coords, zooms=[ZOOM_LEVEL])])   
-        tiles_4326.set_crs(epsg=4326, inplace=True)
-        tiles_4326_all.append(tiles_4326)
-    tiles_4326_aoi = gpd.GeoDataFrame(pd.concat(tiles_4326_all, ignore_index=True))
-
-    # Remove unrelevant tiles and reorganized the data set:
-    logger.info('- Remove duplicated tiles and tiles that are not intersecting labels') 
-
-    # - Keep only tiles that are actually intersecting labels
-    labels_4326.rename(columns={'FID': 'id_aoi'}, inplace=True)
-    tiles_4326 = gpd.sjoin(tiles_4326_aoi, labels_4326, how='inner')
+    # - Remove useless columns, reset feature id and redefine it according to xyz format  
+    logger.info('- Add tile IDs and reorganise the data set')
+    tiles_4326_all_gdf = tiles_4326_all_gdf[['geometry', 'title', 'year'] if 'year' in tiles_4326_all_gdf.keys() else ['geometry', 'title']].copy()
+    tiles_4326_all_gdf.reset_index(drop=True, inplace=True)
+    tiles_4326_all_gdf = tiles_4326_all_gdf.apply(add_tile_id, axis=1)
 
     # - Remove duplicated tiles
     if nb_labels > 1:
```

Comment:
Why check for the number of labels here? As I understand here, this duplicate drop is only to have the right number of tiles printed and limit processing time in `sjoin`. The one created by duplication with the labels are removed later anyways.